src/python/WMCore/Services/DDM/DDM.py

```diff
@@ -0,0 +1,286 @@
+        :api:    The DDM api to be queried.
+        :return: Dictionary containing the result from DDM
+        """
+
+        # try to figure out the api from the ddmReq template
+        if not api:
+            api = ddmReq.api
+
+        if not api:
+            msg = "WARNING: skip sending data to DDM: "
+            msg += "API: {}: ddmReq: {}\n".format(ddmReq.api, pformat(ddmReq))
+            return None
+
+        apiUrl = "registry/request/{}".format(api)
+        callname = "DDM-{}".format(api)
+
+        result = None
+        try:
+            # TODO: here/getResult to send the request parameters to the logger
+            result = self._getResult(ddmReq, apiUrl, callname)
+        except Exception as e:
+            msg = "ERROR: sending data to DDM: \n"
+            msg += "API: {}\n".format(ddmReq.api)
+            msg += "ddmReq: {}\n".format(pformat(ddmReq))
+            msg += "ERROR: {}".format(str(e))
+            self['logger'].debug(msg)
+            return None
+        return result
+
+    def makeAggRequests(self, ddmReqList, aggKey=None):
+        """
+        _makeAggRequest_
+
+        Makes several Aggregated requests to DDM. Takes the list of objects,
+        instances of ddmReqTemplate and aggregates it based on a selected key.
+        It iterates through the list and compares all to all. Uses the
+        ddmReqTemplate internal Equivalence method for comparison with the
+        aggregation key excluded. If a match is found the two requests are
+        squeezed into one. The final result should be a new list of unique
+        ddmRequests each one of them should contain all similar ddmRequests
+        from the initial list with the values of the aggregation key accumulated
+        in a list under the same key name. Creates a request per ddmReqTemplate
+        from the list with aggregated templates.
+        NOTE:
+            In this method the api is always fetched from the template.
+        TODO:
+            So far it is working basically for the 'copy' API and due to that
+            the aggregation key is always a dictionary so no explicit check is
+            done for the type. In case we decide to use it we should develop
+            this to the end with the proper type checks.
+
+        :ddmReqList: List of objects, instances of DDMReqTemplate class.
+        :aggKey:    String representing the Aggregation key.
+        :return:     List of Dictionaries containing the results from DDM
+
+        """
+        if not isinstance(ddmReqList, list) or len(ddmReqList) == 0:
+            return None
+
+        ddmReqListAgg = []
+
+        # populate the first element in the aggregated list
+        ddmReqListAgg.append(ddmReqList.pop())
+
+        # feed the rest
+        while len(ddmReqList) != 0:
+            ddmReq = ddmReqList.pop()
+            for aggReq in ddmReqListAgg:
+                if ddmReq.isEqual(aggReq, aggKey):
```

Comment:
If objects are equal, doesn't it mean that all key/value pairs are the same? If so, we can simply drop `ddmReq` and move to the next object.

Reply:
This is exactly what is happening few lines bellow, but before we drop the object, we need to add the values from the aggregation key of the object under check to the list of values in the aggregation key of the object that it matched.

Reply:
I meant, if objects are **equal**, there is no aggregation to be done, objects are duplicate and we only need to keep one.

Reply:
No, it is not true in the current case. In the current case the equality is checked through the `isEqual()` internal function, which returns `True` if the objects are similar on all other fields/keys, but the excluded one. They differ on that one, and this is the aggregation key in our case. And that's why I needed to develop this method at first place so that we can have a match like that:  Two objects are announced equal if they match on all other but the aggregation key, then we merge the values from the aggregation key in the new one and drop the old object.